samples/net/lwm2m_client/src/lwm2m-client.c

```diff
@@ -344,8 +346,34 @@ static void rd_client_event(struct lwm2m_ctx *client,
 	}
 }
 
+static void event_iface_up(struct net_mgmt_event_callback *cb,
+			   u32_t mgmt_event, struct net_if *iface)
+{
+	int ret;
+
+#if defined(CONFIG_NET_IPV6)
+	ret = lwm2m_rd_client_start(&client, CONFIG_NET_APP_PEER_IPV6_ADDR,
```

Comment:
Please note that the stack might be quite small (CONFIG_NET_MGMT_EVENT_STACK_SIZE, default size is 512 bytes) if event_iface_up() is called from event mgmt thread. This is why I did not call the processing function directly in subsys/net/ip/dhcpv4.c. Have you made sure that lwm2m_rd_client_start() can be called from such a small stack?

Reply:
I was debugging this issue a few days ago.  And you are quite right.  Under normal circumstances, this works just fine.  But with DTLS enabled there are a few too many nested calls in the networking layer and I end up with a stack crash.  How are other samples doing this?

Reply:
One option is to use system workqueue from event handler as then the stack will be much larger. So search k_delayed_work_submit() from samples/net to see some examples.

Reply:
Make sense.  I can re-implement here with a delayed work.

Reply:
@jukkar I worked on this today and adding the delayed_work here in the sample seems.. very fragile.   Instead, I'm going to add the delayed_work to the "start" function in the LwM2M subsys (making sure we're on the main stack no matter how it's called).  Otherwise, issues like this could crop up unexpectedly for users.

My other thought was to add the interface event handling to the subsys as well.  This way less and less code would be needed in the sample to produce reliable results.

Reply:
Ok, makes sense.